subworkflows/local/gene_exon_outrider/main.nf

```diff
@@ -0,0 +1,90 @@
+
+include { OUTRIDER as OUTRIDER_EXON        } from '../../../modules/local/outrider/main'
+include { OUTRIDER as OUTRIDER_GENE        } from '../../../modules/local/outrider/main'
+
+workflow GENE_EXON_OUTRIDER {
+    take:
+    gene_counts // gene expression counts
+    exon_counts // exon expression counts
+    ch_gtf
+
+    main:
+
+
+    // ---------- helpers ----------
+    // build channel with reference panels for each level ('gene'|'exon') and condition ('chx'|'cntrl')
+    def get_refset = { String level, String cond ->
+        parents = params."outrider_${level}_${cond}_parents"
+        index   = params."outrider_${level}_${cond}_index"
+
+        Channel.fromPath("${parents}/*")
+            .collect()
+            .map { ref -> [[id: "${cond}_parents"], ref] }
+            .concat(
+                Channel.fromPath("${index}/*")
+                    .collect()
+                    .map { ref -> [[id: "${cond}_index"], ref] }
+            )
+    }
+
+    // Split CHX/control, group counts per treatment, combine them with ref panels
+    def outriderInput = { countsCh, refsChx, refsCntrl ->
+        def split = countsCh
+            .map { meta, counts -> counts }
+            .branch { v ->
+
+                def s = v.toString().toLowerCase()
+                def isChx   = s.contains('chx')
+                def isCntrl = s.contains('control') || s.contains('cntrl') || s.contains('ctrl')
```

Comment:
Is it possible to have a parameter in nextflow.config that defines these regex strings? Would be easy to change in future if needed.